packages/api/src/platforms/vtex/utils/cookies.ts

```diff
@@ -1,8 +1,84 @@
 import type { Context } from '../index'
 
+const MATCH_FIRST_SET_COOKIE_KEY_VALUE = /^([^=]+)=([^;]*)/
+
+/**
+ * This function updates the ctx.storage.cookies, that is used in each request.
+ *
+ * ctx.storage.cookies is a Map<string[1], Record<string, string>[2]> where,
+ * [1] cookie key
+ * [2] {
+ *       value: cookie value,
+ *       setCookie: setCookie used in browser response
+ *     }
+ */
+const updatesContextStorageCookies = (setCookieValue: string, ctx: Context) => {
+  const matchCookie = setCookieValue.match(MATCH_FIRST_SET_COOKIE_KEY_VALUE)
+  if (matchCookie) {
+    const cookieKey = matchCookie[1]
+    const cookieValue = matchCookie[2]
+
+    ctx.storage.cookies.set(cookieKey, {
+      value: cookieValue,
```

Comment:
I don't think we should save this separately. `key:setCookieValue` would be enough, and then on `updatesCookieValueByKey` you can match with exactly this regex (as I suggested in the other comment) and use the value directly there.